packages/console/src/console-history/ConsoleHistoryItemActions.tsx

```diff
@@ -0,0 +1,81 @@
+/**
+ * Console display for use in the Iris environment.
+ */
+import React, { type ReactElement } from 'react';
+import { Button, CopyButton, ContextualHelp } from '@deephaven/components';
+import './ConsoleHistoryItem.scss';
+import { vsDebugRerun } from '@deephaven/icons';
+import { type ConsoleHistoryActionItem } from './ConsoleHistoryTypes';
+import ConsoleHistoryItemTooltip from './ConsoleHistoryItemTooltip';
+
+interface ConsoleHistoryItemProps {
+  item: ConsoleHistoryActionItem;
+  handleCommandSubmit: (command: string) => void;
+  lastItem?: boolean;
+  firstItem?: boolean;
+  handleTooltipVisible: (isVisible: boolean) => void;
+}
+
+const getActionBarClasses = (
+  command: string | undefined = '',
+  firstItem = false,
+  lastItem = false
+): string => {
+  const lineCount = command.split('\n').length;
+  const classes = ['console-history-actions'];
+
+  if (lineCount === 1) {
+    if (firstItem) {
+      // first single items are pushed down so that they are visible
+      // this should be higher priority than lastItem
+      classes.push('console-history-first-single-line');
+    } else if (lastItem) {
+      // last single items are pushed up to prevent layout shifts
+      classes.push('console-history-last-single-line');
+    } else {
+      classes.push('console-history-single-line');
+    }
+  } else if (lineCount === 2) {
+    classes.push('console-history-two-lines');
+  }
+  return classes.join(' ');
+};
+
+function ConsoleHistoryItem(props: ConsoleHistoryItemProps): ReactElement {
+  const {
+    item,
+    handleCommandSubmit,
+    firstItem,
+    lastItem,
+    handleTooltipVisible,
+  } = props;
+
+  const actionBarClasses = getActionBarClasses(
+    item.command,
+    firstItem ?? false,
+    lastItem ?? false
+  );
```

Comment:
If possible I'd rather write this using the `classnames` package, e.g.

```
import classNames from 'classnames';

...
return <div className={classNames('console-history-actions', `console-history-actions-line-${lineCount}`)}> ...
```
Also then you can just do CSS for `console-history-actions-line-1` and `console-history-actions-line-2` etc.